File: CSES/Sorting/FerrisWheel.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
#define fast ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

int main() {
    fast;
    int n, x;
    cin >> n >> x;
    vector<int> nums(n);
    for(int i = 0; i < n; i++)
        cin >> nums[i];
    sort(nums.begin(), nums.end());
    int left = 0, right = n-1, result = 0;
    while(left <= right) {
        if(left <= right && nums[left]+nums[right] <= x) {
            left++;
            right--;
        } else
            right--;
        result++;
    }
    cout << result;
    return 0;
}

File: CSES/Sorting/FerrisWheel.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class FerrisWheel {
    public static class FastReader {
        public BufferedReader buffer;
        public StringTokenizer tokenizer;

        public FastReader() {this.buffer = new BufferedReader(new InputStreamReader(System.in));}

        public String next() {
            while(tokenizer == null || !tokenizer.hasMoreTokens()) {
                try {tokenizer = new StringTokenizer(buffer.readLine());}
                catch(IOException e) {e.printStackTrace();}
            }
            return tokenizer.nextToken();
        }

        public int nextInt() {return Integer.parseInt(next());}
        public long nextLong() {return Long.parseLong(next());}
    }

    public static void main(String[] args) {
        FastReader fast = new FastReader();
        final int n = fast.nextInt(), x = fast.nextInt();
        int nums[] = new int[n];
        for(int i = 0; i < n; i++)
            nums[i] = fast.nextInt();
        System.out.print(solve(n, x, nums));
    }

    public static int solve(final int n, final int x, int nums[]) {
        Arrays.sort(nums);
        int left = 0, right = n-1, ans = 0;
        while(left <= right) {
            if(left < right && nums[left] + nums[right] <= x) {
                left++;
                right--;
            } else
                right--;
            ans++;
        }
        return ans;
    }
}
